File: name-score/name-score-rest-server/src/main/java/com/shrader/namescore/parser/StreamParser.java
package com.shrader.namescore.shell.scoring.parser;

import org.springframework.stereotype.Component;

import java.nio.CharBuffer;
import java.util.List;
import java.util.regex.Pattern;
import java.util.stream.Collectors;


@Component
public class StreamParser implements NameParser<String> {

    @Override
    public List<String> parse(CharBuffer charBuffer, String delimiter) {
        return  Pattern.compile(delimiter)
                .splitAsStream(charBuffer)
                .peek(e -> System.out.println("Original value: " + e))
                .map(e -> e.replaceAll("\"", ""))
                .peek(e -> System.out.println("Mapped value: " + e))
                .map(e -> e.trim())
                .peek(e -> System.out.println("Mapped value: " + e))
                .collect(Collectors.toList());
    }
}

File: name-score/name-score-shell/src/main/kotlin/com/shrader/namescore/shell/parser/StreamParser.kt
package com.shrader.namescore.shell.parser

import org.springframework.stereotype.Component
import java.nio.CharBuffer
import java.util.regex.Pattern
import java.util.stream.Stream
import kotlin.streams.toList


@Component
class StreamParser : NameParser<String> {

    override fun parse(charBuffer: CharBuffer, delimiter: String): List<String> {
        val stream: Stream<String> = Pattern.compile(delimiter).splitAsStream(charBuffer)

        return stream
                .peek { println("Original value: $it") }
                .map { it.replace("\"", "") }
                .peek { println("Original value: $it") }
                .map { it.trim() }
                .peek { println("Mapped value: $it") }
                .toList<String>()
    }
}